brightlayer-ui/react-design-patterns
# add firebase workflows and deploy

Fixes # .


#### Changes proposed in this Pull Request:
- add firebase workflows: firebase-hosting-pull-request and firebase-hosting-merge
- firebase-hosting-pull-request yml contains the set expire 2 days. (dev workflow site to test)
- firebase-hosting-merge yml is set to master branch. Workflow kicks off on master merge then deploys it

Firebase hosting via GitHub actions creates a preview channel (URL) on every pull request. As the pull request cycles through its changes the deployment also updates.
The preview channel is a temporary channel (URL) that has a default expiration time of 2 days and the expiration (days) can be set to anything.
The GitHub action also includes a second workflow that will deploy a fresh site to a set channel (URL) once the feature branch is merged to master branch.

This is a free plan (spark plan) includes testing, analytics, app distribution and app indexing at no cost across all plans. Cloud store service (included in spark plan) includes 10GB total storage , document writes are 20k per day and document reads are 50k per day. Custom domain names with SSL is included as well.

#### Screenshots:


#### To Test:
-  commit a basic change to this PR. The action workflow kicks off and updates the current temp deployment
- after merge happens, navigate to the actions tab in this repo. You will see the master deploy workflow run. Within that workflow you should see a step like this -
- 
![image](https://user-images.githubusercontent.com/37150565/167179031-be982c73-be08-4b1f-9911-3ebf308b2300.png)
